generate-out-dir.py

```
@@ -0,0 +1,255 @@
+        # Setup logging
+        setup_logging()
+
+        # Load environment variables
+        load_dotenv()
+
+        # Get base output directory and clustering algorithm
+        base_output_dir = Path(os.getenv("OUTPUT_DIR_PATH", ""))
+        algorithm = os.getenv("CLUSTERING_ALGORITHM")
+
+        if not base_output_dir:
+            raise ValueError("OUTPUT_DIR_PATH environment variable is required")
```

Ruff (0.11.9)

Abstract `raise` to an inner function

(TRY301)